skops/card/_model_card.py

```diff
@@ -1,61 +1,120 @@
-
-    Parameters:
-    ----------
-    model: estimator
-        scikit-learn compatible estimator.
-    card_data: CardData
-        CardData object.
-    card_kwargs:
-        Card kwargs are information you can pass to fill in the sections of the
-        card template, e.g. model_description, citation_bibtex, get_started_code.
-    """
-    ROOT = skops.__path__
-    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
-    hyperparameter_table = _extract_estimator_config(model)
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
+        self.model = model
+        self.hyperparameter_table = self._extract_estimator_config()
+        # the spaces in the pipeline breaks markdown, so we replace them
+        self.model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+        self.template_sections = {}
+        self.figure_paths = {}
+
+    def add(self, **kwargs):
+        """Takes values to fill model card template.
+        Parameters:
+        ----------
+        **kwargs : dict
+            Parameters to be set for the model card. These parameters
+            need to be sections of the underlying `jinja` template used.
+
+        """
+        for section, value in kwargs.items():
+            self.template_sections[section] = value
+        return self
+
+    def add_inspection(self, **kwargs):
+        """Takes plots to fill model card template.
+        Parameters:
+        ----------
+        Parameters to be set for the model card. These parameters
+        need to be sections of the underlying `jinja` template used.
+        """
+        for plot_name, plot_path in kwargs.items():
+            self.figure_paths[plot_name] = plot_path
+        return self
+
+    def save(self, path):
+        """Fills model card template, renders and saves it as markdown file to the target directory.
+
+        Parameters:
+        ----------
+        path: Path
+              filepath to save your card.
+        """
+        ROOT = skops.__path__
+
+        metadata_keys = [
+            "language",
+            "license",
+            "library_name",
+            "tags",
+            "datasets",
+            "model_name",
+            "metrics",
+            "model-index",
+        ]
+        card_data_keys = {}
+
+        # if key is supposed to be in metadata and is provided by user, write it to card_data_keys
+        for key in metadata_keys:
+            if key in self.template_sections.keys():
+                card_data_keys[key] = self.template_sections.pop(key, "")
```

Comment:
I would avoid mutating the `self.template_section`. As an alternative, I would suggest to create a copy at the beginning of the method and perform all operations on that copy:

```python
template_sections = self.template_sections.copy()
...
template_sections.pop(key)
```

Right now, if I call the `save` method twice, the 2nd time around, the `template_sections` will not contain any meta data entries anymore, right?